### PR TITLE
Correct window bandwidth calculation and documentation

### LINKED ISSUE #1527: Where does the formula in `window_bandwidth` come from?
**Is your feature request related to a problem? Please describe.**
Here's our formula for `window_bandwidth`

```python
        win = get_window(window, n)
        WINDOW_BANDWIDTHS[key] = n * np.sum(win ** 2) / np.sum(np.abs(win)) ** 2
```

This formula is (proportional to) the ratio between the square of the L2 norm and the square of the L1 norm.
I see how this relates to the (inverse) "peakiness" of the window vector, but i'd like some mathematical context around it.

**Describe the solution you'd like**
Some explanation in the docstring as to why we use this formula + where it comes from. Maybe a bibliographical /textbook reference.

**Describe alternatives you've considered**
* Knowledge of the First Kind
* Knowledge of the Third Kind

https://www.gutenberg.org/files/3800/3800-h/3800-h.htm

**Additional context**
Found while discussing #745

### PR DESCRIPTION
**Pull request by bmcfee**
#### Reference Issue
Fixes #1527 

#### What does this implement/fix? Explain your changes.

This PR fixes a very small error in `filters.window_bandwidth`, and expands the documentation.

Only window functions with negative values are affected.

#### Any other comments?

This should be good to merge after CI.

### CODE REVIEW

File: librosa/filters.py [hunk 1]
```diff
@@ -1115,7 +1124,7 @@ def window_bandwidth(window, n=1000):
 
     if key not in WINDOW_BANDWIDTHS:
         win = get_window(window, n)
-        WINDOW_BANDWIDTHS[key] = n * np.sum(win ** 2) / np.sum(np.abs(win)) ** 2
+        WINDOW_BANDWIDTHS[key] = n * np.sum(win ** 2) / (np.sum(win) ** 2 + util.tiny(win))
```
Comment (bmcfee) [inline librosa/filters.py:1127]: Just a note that I'm adding a tiny here to stabilize the division.  This shouldn't affect anything substantively —I'd be worried if a window sums to an underflow value!—but it makes me feel better.

**Comment by lostanlen** [comment]
thanks. does the Harris paper have a proof that this formula indeed yields "the width of an equivalent ideal rectangular spectral response that will pass the same noise power as the window (filter) under test" ?

**Comment by bmcfee** [comment]
> does the Harris paper have a proof that this formula indeed yields

It's not quite as streamlined as the other link that I provided in #1527 discussion, but yes:
image

**Review by lostanlen — COMMENTED** [review]
Thank you

**Pull request merged**